src/dc_receive_imf.rs

```diff
@@ -1414,6 +1444,79 @@ async fn create_or_lookup_group(
     Ok((chat_id, chat_id_blocked))
 }
 
+#[allow(clippy::indexing_slicing)]
+async fn create_or_lookup_mailinglist(
+    context: &Context,
+    allow_creation: bool,
+    create_blocked: Blocked,
+    list_id_header: &str,
+    subject: &str,
+) -> (ChatId, Blocked) {
+    static LIST_ID: Lazy<Regex> = Lazy::new(|| Regex::new(r"^(.*.)<(.*.)>$").unwrap());
+    let (mut name, listid) = match LIST_ID.captures(list_id_header) {
+        Some(cap) => (cap[1].trim().to_string(), cap[2].trim().to_string()),
+        None => (
+            "".to_string(),
+            list_id_header
+                .trim()
+                .trim_start_matches('<')
+                .trim_end_matches('>')
+                .to_string(),
+        ),
+    };
+
+    if let Ok((chat_id, _, blocked)) = chat::get_chat_id_by_grpid(context, &listid).await {
+        return (chat_id, blocked);
+    }
+
+    static SUBJECT: Lazy<Regex> = Lazy::new(|| Regex::new(r"^.{0,5}\[(.*.)\]").unwrap());
+    if let Some(cap) = SUBJECT.captures(subject) {
+        name = cap[1].to_string();
```

Comment:
Because in the few mailinglist examples that I saw the List-Id was way too long, e.g. on the DC mailing list `"discussions about and around https://delta.chat developments"` (subject topic is `delta-dev`). Also, people will probably choose the subject more carefully as it is more annoying if it's unfitting or too long.

Reply:
k, that sounds reasonable.